Formula/chapel.rb

```diff
@@ -12,23 +12,36 @@ class Chapel < Formula
     sha256 x86_64_linux: "ca34ea32e25b7c9fea13bf25a09ac8b3151f45f8b8ac2dfeebfc0d18773783ba"
   end
 
+  depends_on "llvm@11"
   depends_on "python@3.9"
 
   def install
     libexec.install Dir["*"]
     # Chapel uses this ENV to work out where to install.
     ENV["CHPL_HOME"] = libexec
     # This is for mason
-    ENV["CHPL_REGEXP"] = "re2"
+    ENV["CHPL_RE2"] = "bundled"
 
     # Must be built from within CHPL_HOME to prevent build bugs.
     # https://github.com/Homebrew/legacy-homebrew/pull/35166
     cd libexec do
+      system "./util/printchplenv", "--all"
+      system "make"
+      # Need to let chapel choose target compiler with llvm
+      ENV["CHPL_HOST_CC"] = ENV["CC"]
+      ENV["CHPL_HOST_CXX"] = ENV["CXX"]
+      ENV.delete("CC")
+      ENV.delete("CXX")
```

Comment:
Why delete these?

Reply:
We found an 11th hour bug in our Makefile system that caused system clang rather than homebrew's llvm@11 clang to be used as the C compiler for certain build steps, so making these settings worked around the bug for now; we expect to address this in our next Chapel release to avoid the workaround.  https://github.com/chapel-lang/chapel/issues/18450 tracks this issue.